rundeckapp/grails-app/services/rundeck/services/PluginApiService.groovy

```diff
@@ -62,24 +71,34 @@ class PluginApiService {
         IFramework framework = frameworkService.getRundeckFramework()
         Locale locale = getLocale()
 
-        //framework level plugin descriptions
-        //TODO: use pluginService.listPlugins for these services/plugintypes
-        Map<String,List<Description>> pluginDescs= [
-                framework.getNodeExecutorService(),
-                framework.getFileCopierService(),
-                framework.getNodeStepExecutorService(),
-                framework.getStepExecutionService()
-        ].collectEntries{
-            [it.name, it.listDescriptions().sort {a,b->a.name<=>b.name}]
-        }
-
-        //load via pluginService to include spring-based app plugins
-        pluginDescs['ResourceModelSource'] = pluginService.listPlugins(
-                ResourceModelSourceFactory,
-                framework.getResourceModelSourceService()
-        ).findAll { it.value.description }.collect {
-            it.value.description
-        }.sort { a, b -> a.name <=> b.name }
+        Map<String, List<Description>> pluginDescs = [:]
+
+        StepExecutionService ses = framework.getStepExecutionService()
+        pluginDescs[ses.name] = pluginService.listPlugins(StepExecutor, ses)
+                .findAll { it.value.description != null }
```

Comment:
Note the are still a few plugin services that contain plugins descriptors (`DescribedPlugin`) without a description property. The previous `service.listDescriptions()` approach filtered these out. The  `.findAll` is present here to compensate.